### PR TITLE
docs: Initial docs and JupyterBook setup

### PR DESCRIPTION
**Pull request by ayushnag**
This pull request includes several changes to improve the documentation setup and configuration for the `phylo2vec` project. The changes include adding configuration files for Read the Docs and Jupyter Book, updating the table of contents and API reference, and adding necessary dependencies.

Documentation setup and configuration:

* [`.readthedocs.yaml`](diffhunk://#diff-03efc769b870804394632e45d7885272b44c16939517fb31c9d7c614d2ffae57R1-R23): Added a configuration file for Read the Docs to specify the build environment and tools required for generating documentation.
* [`docs/_config.yml`](diffhunk://#diff-f64371f310bcd3bbf6ead5827d929a4eb13b7516806d6658485a996e39eb2556R1-R39): Added a configuration file for Jupyter Book to set book settings, execution options, repository information, and Sphinx extensions.
* [`docs/_toc.yml`](diffhunk://#diff-4f938c2f382334bbe68d17da3c7582b77496a7bf3a4d6c2ff53d1c073500e163R1-R7): Added a table of contents file for Jupyter Book to define the structure of the documentation.
* [`docs/api.rst`](diffhunk://#diff-0ee295c29def16f570f86378eff8ba37315e464d415e731d5d8ea2e67ce16f68R1-R44): Added an API reference file to document the functions and modules in the `phylo2vec` project using Sphinx autosummary.
* [`docs/requirements.txt`](diffhunk://#diff-2c9c11d26b09b8afde329980309d967121543a456e4592c76886a20b5cf56c90R1): Added a requirement for `jupyter-book` to ensure the necessary dependencies are installed.

Additionally, the `pixi.toml` file was updated to include a task for building the documentation using Jupyter Book:

* [`pixi.toml`](diffhunk://#diff-b092682f0ad15a352f9fa2b0a67c992454b586275ee76b74a5839ab9afdf9894R63-R70): Added a `build-docs` task to the `feature.python.tasks` section to automate the documentation build process.

### CODE REVIEW

File: .github/workflows/documentation-links.yml
Comment (lsetiawan) [inline .github/workflows/documentation-links.yml:1]: I activated this on the readthedocs side, no need for PR previews. Please remove. 😄 Thanks!

**Comment by lsetiawan** [comment]
Looks like the build is failing

**Comment by lsetiawan** [comment]
Are you able to see the build logs btw?

**Comment by ayushnag** [comment]
Yes I can see the RTD previews and the log outputs. Somehow `jupyter-book` is not importing even though it's added to the `pixi.toml`. I'll take a look

**Comment by lsetiawan** [comment]
This is because RTD is not using pixi toml to actually install anything, you'll have to modify the `.readthedocs` to make it so that pixi is used

**Comment by lsetiawan** [comment]
See https://github.com/readthedocs/readthedocs.org/issues/11548#issuecomment-2514917065 for related conversation about this

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+build:
+  os: ubuntu-22.04
+  tools:
+    python: "3.11"
+  jobs:
+    install:
+      - asdf plugin add pixi
+      - asdf install pixi latest
+      - asdf global pixi latest
+      - pixi run build-docs
```
Comment (lsetiawan) [inline .readthedocs.yaml:18]:
```suggestion
      - pixi run -e py-phylo2vec build-docs
```

File: pixi.toml [hunk 2]
```diff
@@ -60,10 +60,14 @@ pip = ">=24.2,<25"
...
 jupyterlab = ">=4.1,<4.2"
 pytest = ">=8.3.4,<9"
+jupyter-book = ">=0.11.0"
 
 [feature.python.tasks.install-python]
 cmd = "maturin develop --manifest-path ./py-phylo2vec/Cargo.toml"
 
+[feature.python.tasks.build-docs]
+cmd = "jupyter-book build docs"
+
```
Comment (lsetiawan) [inline pixi.toml:70]:
```suggestion
[feature.python.tasks.build-docs]
cmd = "jupyter-book build docs"
depends-on = ["install-python"]

```

**Comment by lsetiawan** [comment]
@ayushnag It works now: https://uw-phylo2vec--66.org.readthedocs.build/en/66/

**Comment by ayushnag** [comment]
Thanks for the help @lsetiawan!

**Comment by lsetiawan** [comment]
No problem! Thanks for setting it up! 😄

**Comment by lsetiawan** [comment]
Good to know we can use pixi to setup jupyterbook!

**Review by lsetiawan — APPROVED** [review]
LGTM

File: docs/requirements.txt
Comment (lsetiawan) [inline docs/requirements.txt:1]: Could you remove this actually?

Comment (ayushnag) [reply docs/requirements.txt:1]: Done

**Pull request merged**